### ISSUE TITLE
Apakah Bottleneck Performa dalam Aplikasi Go Lebih Dipengaruhi oleh Database dan I/O?

### ISSUE THREAD

**Issue by ilhamgepe** [ProgrammerZamanNow/qna#672]
halo mas eko dan semuanya!

setelah baca artikel dan diskusi, performa **go stdlib/framework** sudah cukup cepat dalam banyak kasus. 
aku berasumsi memilih framework yang tidak terlalu cepat sekalipun eg:(_chi,echo_) tidak menjadi bottleneck utama dalam aplikasi Go. 
tapi bottleneck utama dalam performa aplikasi Go, lebih sering berada di lapisan **database atau I/O**. Dalam konteks ini, apa bener ketika kita mengoptimasi aplikasi Go, kita lebih banyak harus fokus pada optimasi query database atau pengelolaan I/O daripada memikirkan apakah framework yang kita pilih bisa memberikan performa lebih baik?

1 lg nambah hehe. 
btw kalo microservice itu 1 service kan db nya terpisah ya(_bener2 belum pernah microservice_)?
aku bingung soalnya di aplikasi pasti kan ada relasi, nah itu gimana ya. aku tanya chat gpt masih belum paham, yang aku paham sejauh ini.
1. 1 db perservice
2. relasi bisa di akalin pake event-driven(kata chat gpt pake aplikasi apa gitu lupa), misal ketika user update, kita kirim juga data updatenya itu ke db lain yang punya relasi ke user. jadi datanya tetep konsisten antar service
apakah bener begitu?

trimakasih

**Comment by bagasdisini** [comment]
Yes bener, di bahasa Go, pemilihan framework itu bukan permasalahan utama. Mau pake framework apapun, atau bahkan ga pake itu gaada problem sama sekali. Di Go itu gaada framework yang 'gede' banget kaya misal Laravel atau Django. Jadi fokus aja ke optimasi query dan I/O.

**Comment by khannedy** [comment]
https://youtu.be/mzv9xz1WOFE
